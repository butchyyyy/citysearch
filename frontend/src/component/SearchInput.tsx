import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
import React from "react"
import { Input, InputGroup, InputGroupAddon, InputGroupText } from "reactstrap"

interface Props {
  /** Actual search input value */
  value: string
  /** Method handling search submit */
  onSubmitSearch: () => void
  /** Method handling search input */
  onSearchInput: (query: string) => void
}

/**
 * Component rendering user city search input p
 */
const SearchInput = (props: Props) => {
  /** Handle user input change */
  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    props.onSearchInput(e.target.value)
  }

  /** Hender 'Emter' keypress input, that submits search */
  const handleKeyPress = (e: React.KeyboardEvent) => {
    if (e.key === "Enter") {
      props.onSubmitSearch()
    }
  }

  return (
      <InputGroup>
        <InputGroupAddon addonType="prepend">
          <InputGroupText>
            <FontAwesomeIcon icon="search-location" />
          </InputGroupText>
        </InputGroupAddon>
        <Input value={props.value} onChange={handleChange} onKeyPress={handleKeyPress} placeholder="Enter City Name, Maximum of 1000 results will be shown" />
      </InputGroup>
  )
}

export default SearchInput
