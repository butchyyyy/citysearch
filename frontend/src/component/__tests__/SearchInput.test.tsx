import SearchInput from "component/SearchInput"
import { shallow, ShallowWrapper } from "enzyme"
import React from "react"
import { Input } from "reactstrap"

describe("Test SerachInput Component", () => {
  const submitSearchMock = jest.fn()
  const searchInputMock = jest.fn()
  const wrapper = shallow(<SearchInput value="test" onSubmitSearch={submitSearchMock} onSearchInput={searchInputMock} />)

  it("Test SerachInput Snapshot", () => {
    expect(wrapper).toMatchSnapshot()
  })

  it("Test Search Submit", () => {
    const input: ShallowWrapper = wrapper.find(Input)
    input.simulate("keyPress", { key: "a" })
    expect(submitSearchMock).toHaveBeenCalledTimes(0)
    input.simulate("keyPress", { key: "Enter" })
    expect(submitSearchMock).toHaveBeenCalled()
  })

  it("Test Search Input", () => {
    const input: ShallowWrapper = wrapper.find(Input)
    input.simulate("change", { target: { value: "newValue" } })
    expect(searchInputMock).toHaveBeenCalledWith("newValue")
  })
})
