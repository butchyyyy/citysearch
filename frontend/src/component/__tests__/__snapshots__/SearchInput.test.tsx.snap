// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SerachInput Component Test SerachInput Snapshot 1`] = `
<InputGroup
  tag="div"
>
  <InputGroupAddon
    addonType="prepend"
    tag="div"
  >
    <InputGroupText
      tag="span"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="search-location"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
    </InputGroupText>
  </InputGroupAddon>
  <Input
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Enter City Name, Maximum of 1000 results will be shown"
    type="text"
    value="test"
  />
</InputGroup>
`;
